src/Fmt_ast.ml

```diff
@@ -606,11 +606,29 @@ let fmt_docstring c ?(standalone = false) ?pro ?epi doc =
         | Some (_, false), true -> true
         | _, false -> false
       in
+      let try_parse str_cmt =
+        let if_fail = if c.conf.wrap_comments then fill_text else str in
+        match Octavius.parse (Lexing.from_string str_cmt) with
+        | Error _ -> if_fail str_cmt
+        | Ok parsed ->
+            if Conf.debug then (
+              Octavius.print Caml.Format.str_formatter parsed ;
+              Caml.Format.eprintf "%s%!"
+                (Caml.Format.flush_str_formatter ()) ) ;
+            let fmted = Fmt_odoc.fmt parsed in
+            let spaces = ['\t'; '\n'; '\011'; '\012'; '\r'; ' '] in
+            let is_space = List.mem ~equal:Char.equal spaces in
+            let space_0 = try is_space str_cmt.[0] with _ -> false in
+            let space_n =
+              try is_space str_cmt.[String.length str_cmt - 1] with _ ->
+                false
```

Comment:
Please don't rely on exceptions to avoid out of bound array accesses. Setting up the handler is (comparatively) slow, and it is not unheard of to compile with noassert.

Are the only exceptions here due to array indexing?